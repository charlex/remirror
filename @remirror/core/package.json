{
  "name": "@remirror/core",
  "description": "Provides the core abstractions for managing your remirror editor functionality",
  "license": "MIT",
  "author": "Ifiok Jr. <ifiokotung@gmail.com>",
  "homepage": "https://github.com/ifiokjr/remirror/tree/master/@remirror/core",
  "repository": "https://github.com/ifiokjr/remirror/tree/master/@remirror/core",
  "version": "0.4.2-canary.2",
  "main": "lib/index.js",
  "module": "lib/dist/core.esm.js",
  "browser": {
    "min-document": false
  },
  "files": [
    "internal",
    "lib",
    "src"
  ],
  "scripts": {
    "api": "api-extractor run",
    "api:local": "api-extractor run --local",
    "build": "run-p build:*",
    "build:babel": "babel src --out-dir lib --root-mode=upward --extensions \".ts,.tsx\" -s",
    "build:declaration": "tsc -p ./tsconfig.prod.json --emitDeclarationOnly --declarationMap",
    "dev": "run-p dev:*",
    "dev:babel": "babel src --out-dir lib --root-mode=upward --extensions \".ts,.tsx\" -s --watch",
    "dev:declaration": "tsc -p ./tsconfig.prod.json --emitDeclarationOnly --declarationMap --watch",
    "document": "node ../api-documenter/lib/start.js markdown --input-folder temp --output-folder ../../docs/api/@remirror",
    "lint": "tslint --project tsconfig.lint.json --config ../../tslint.json",
    "typecheck": "tsc -p ./tsconfig.json --noEmit"
  },
  "types": "lib/index.d.ts",
  "dependencies": {
    "@babel/runtime": "^7.6.0",
    "@remirror/core-constants": "0.4.2-canary.2",
    "@remirror/core-helpers": "0.4.2-canary.2",
    "@remirror/core-types": "0.4.2-canary.2",
    "@remirror/core-utils": "0.4.2-canary.2",
    "@remirror/react-portals": "0.4.2-canary.2",
    "@types/prosemirror-inputrules": "^1.0.2",
    "@types/prosemirror-keymap": "^1.0.1",
    "@types/prosemirror-model": "^1.7.2",
    "@types/prosemirror-state": "^1.2.3",
    "prosemirror-inputrules": "^1.0.4",
    "prosemirror-keymap": "^1.0.1",
    "prosemirror-model": "^1.7.2",
    "prosemirror-state": "^1.2.4",
    "prosemirror-suggest": "0.4.2-canary.2"
  },
  "peerDependencies": {
    "prosemirror-view": "^1.11.1",
    "@types/prosemirror-view": "^1.9.0",
    "@emotion/core": "^10",
    "@types/react": "^16.8.0",
    "react": "^16.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "cjs": "lib/dist/core.cjs.js",
  "meta": {
    "sizeLimit": "55 KB"
  },
  "sideEffects": false
}
