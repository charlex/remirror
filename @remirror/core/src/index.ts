export * from './extension';
export * from './extension-helpers';
export * from './extension-manager';
export * from './extension-types';
export * from './mark-extension';
export * from './node-extension';
export * from './nodes';

export * from '@remirror/core-constants';
export * from '@remirror/core-types';
export * from '@remirror/core-utils';
export * from '@remirror/core-helpers';
