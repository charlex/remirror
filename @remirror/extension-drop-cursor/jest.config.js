const config = require('../../support/jest/jest.config');

module.exports = {
  ...config,
  name: '@remirror/extension-drop-cursor',
  displayName: 'extension-drop-cursor',
};
