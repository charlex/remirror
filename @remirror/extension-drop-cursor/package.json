{
  "name": "@remirror/extension-drop-cursor",
  "description": "CodeBlock extension for the remirror wysiwyg editor",
  "license": "MIT",
  "author": "Ifiok Jr. <ifiokotung@gmail.com>",
  "homepage": "https://github.com/ifiokjr/remirror/tree/master/@remirror/extension-drop-cursor",
  "repository": "https://github.com/ifiokjr/remirror/tree/master/@remirror/extension-drop-cursor",
  "version": "0.4.2-canary.2",
  "main": "lib/index.js",
  "module": "lib/dist/extension-drop-cursor.esm.js",
  "files": [
    "internal",
    "lib",
    "src"
  ],
  "scripts": {
    "_build": "run-p build:*",
    "_typecheck": "tsc -p ./tsconfig.json --noEmit",
    "build:babel": "babel src --out-dir lib --root-mode=upward --extensions \".ts,.tsx\" -s",
    "build:declaration": "tsc -p ./tsconfig.prod.json --emitDeclarationOnly --declarationMap",
    "dev": "run-p dev:*",
    "dev:babel": "babel src --out-dir lib --root-mode=upward --extensions \".ts,.tsx\" -s --watch",
    "dev:declaration": "tsc -p ./tsconfig.prod.json --emitDeclarationOnly --declarationMap --watch",
    "lint": "tslint --project tsconfig.lint.json --config ../../tslint.json"
  },
  "types": "lib/index.d.ts",
  "dependencies": {
    "@babel/runtime": "^7.5.5",
    "@emotion/core": "^10.0.14",
    "@remirror/core": "0.4.2-canary.2",
    "@remirror/react-utils": "0.4.2-canary.2",
    "@types/prismjs": "^1.16.0",
    "@types/prosemirror-commands": "^1.0.1",
    "@types/prosemirror-inputrules": "^1.0.2",
    "@types/prosemirror-keymap": "^1.0.1",
    "@types/prosemirror-state": "^1.2.3",
    "@types/prosemirror-transform": "^1.1.0",
    "@types/prosemirror-view": "^1.9.0",
    "@types/react-syntax-highlighter": "^10.2.1",
    "@types/refractor": "^2.8.0",
    "emotion": "^10.0.14",
    "prosemirror-commands": "^1.0.8",
    "prosemirror-inputrules": "^1.0.4",
    "prosemirror-keymap": "^1.0.1",
    "prosemirror-state": "^1.2.4",
    "prosemirror-transform": "^1.1.3",
    "prosemirror-view": "^1.10.0"
  },
  "peerDependencies": {
    "@types/react": "^16.8.0",
    "react": "^16.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "cjs": "lib/dist/extension-drop-cursor.cjs.js",
  "sideEffects": false
}
