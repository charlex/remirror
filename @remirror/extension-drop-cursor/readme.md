<div align="center">
	<br />
	<div align="center">
		<img width="300" src="https://cdn.jsdelivr.net/gh/ifiokjr/remirror/support/assets/logo-icon.svg" alt="remirror" />
    <h1 align="center">extension-drop-cursor</h1>
	</div>
    <br />
    <br />
    <br />
    <br />
</div>

[![npm bundle size (scoped)](https://img.shields.io/bundlephobia/minzip/@remirror/extension-drop-cursor.svg?style=for-the-badge)](https://bundlephobia.com/result?p=@remirror/extension-drop-cursor) [![npm](https://img.shields.io/npm/dm/@remirror/extension-drop-cursor.svg?style=for-the-badge&logo=npm)](https://www.npmjs.com/package/@remirror/extension-drop-cursor) [![Dependencies (path)](https://img.shields.io/david/ifiokjr/remirror.svg?logo=npm&path=@remirror%2Fextension-drop-cursor&style=for-the-badge)](https://github.com/ifiokjr/remirror/blob/master/@remirror/extension-drop-cursor/package.json) [![NPM](https://img.shields.io/npm/l/@remirror/extension-drop-cursor.svg?style=for-the-badge)](https://github.com/ifiokjr/remirror/blob/master/LICENSE) [![GitHub issues by-label](https://img.shields.io/github/issues/ifiokjr/remirror/@remirror/extension-drop-cursor.svg?label=Open%20Issues&logo=github&style=for-the-badge)](https://github.com/ifiokjr/remirror/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%40remirror%2Fextension-drop-cursor) [![GitHub pull requests by-label](https://img.shields.io/github/issues-pr/ifiokjr/remirror/@remirror/extension-drop-cursor.svg?label=Open%20Pull%20Requests&logo=github&style=for-the-badge)](https://github.com/ifiokjr/remirror/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+sort%3Aupdated-desc+label%3A%40remirror%2Fextension-drop-cursor)

Add a powerful and configurable drop cursor into your editor.
