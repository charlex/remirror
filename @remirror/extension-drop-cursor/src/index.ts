export * from './drop-cursor-plugin';
export * from './drop-cursor-types';
export * from './drop-cursor-extension';
