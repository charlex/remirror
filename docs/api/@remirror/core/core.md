<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@remirror/core](./core.md)

## core package

## Classes

|  Class | Description |
|  --- | --- |
|  [DocExtension](./core.docextension.md) | The parent node in a Prosemirror Schema. A representation of the doc is required in all editors  |
|  [Extension](./core.extension.md) | Extensions are fundamental to the way that Remirror works and they handle the management of similar concerns. |
|  [ExtensionManager](./core.extensionmanager.md) | A class to manage the extensions and prosemirror interactions of our editor. |
|  [MarkExtension](./core.markextension.md) | All mark extensions should be created from here. |
|  [NodeExtension](./core.nodeextension.md) | Defines an abstract class for creating extensions which can place nodes into the prosemirror state. |
|  [TextExtension](./core.textextension.md) | The default text passed into the prosemirror schema. |

## Interfaces

|  Interface | Description |
|  --- | --- |
|  [DocExtensionOptions](./core.docextensionoptions.md) |  |
|  [Extension](./core.extension.md) |  |
|  [ExtensionConstructor](./core.extensionconstructor.md) | An extension constructor |
|  [ExtensionListParams](./core.extensionlistparams.md) |  |
|  [ExtensionManagerData](./core.extensionmanagerdata.md) |  |
|  [ExtensionParams](./core.extensionparams.md) |  |
|  [ManagerParams](./core.managerparams.md) |  |
|  [OnTransactionManagerParams](./core.ontransactionmanagerparams.md) |  |
|  [PrioritizedExtension](./core.prioritizedextension.md) | Provides a priority value to the extension which determines the priority. |

## Variables

|  Variable | Description |
|  --- | --- |
|  [isExtension](./core.isextension.md) | Determines if the passed in extension is a any type of extension. |
|  [isExtensionManager](./core.isextensionmanager.md) | Checks to see whether this is an extension manager |
|  [isMarkExtension](./core.ismarkextension.md) | Determines if the passed in extension is a mark extension. Useful as a type guard where a particular type of extension is needed. |
|  [isNodeExtension](./core.isnodeextension.md) | Determines if the passed in extension is a node extension. Useful as a type guard where a particular type of extension is needed. |
|  [isPlainExtension](./core.isplainextension.md) | Checks whether the this is an extension and if it is a plain one |

## Type Aliases

|  Type Alias | Description |
|  --- | --- |
|  [ActionNames](./core.actionnames.md) | Utility type for pulling all the action names from a list |
|  [ActionsFromExtensions](./core.actionsfromextensions.md) | Utility type which receives an extension and provides the type of actions it makes available. |
|  [AnyExtension](./core.anyextension.md) | Provides a type annotation which is applicable to any extension type. |
|  [AnyExtensionManager](./core.anyextensionmanager.md) | The utility for capturing all the possible <code>ExtensionManager</code> type variations. |
|  [CommandsOfExtension](./core.commandsofextension.md) | Utility type for retrieving the commands provided by an extension. |
|  [DataOfExtension](./core.dataofextension.md) | Utility type for retrieving the extension data object made available from the extension. |
|  [ExtensionsFromManager](./core.extensionsfrommanager.md) | Retrieve the extensions from an <code>ExtensionManager</code>. |
|  [FlexibleExtension](./core.flexibleextension.md) | Either a PrioritizedExtension or the actual Extension. |
|  [FlexibleExtensions](./core.flexibleextensions.md) |  |
|  [FromFlexibleExtension](./core.fromflexibleextension.md) |  |
|  [HelperNames](./core.helpernames.md) | Utility type for pulling all the action names from a list |
|  [HelpersFromExtensions](./core.helpersfromextensions.md) | Utility type which receives an extension and provides the type of helpers it makes available. |
|  [HelpersOfExtension](./core.helpersofextension.md) | Utility type for retrieving the helpers provided by an extension. |
|  [InferFlexibleExtension](./core.inferflexibleextension.md) | Get the extension from a PrioritizedExtension. |
|  [InferFlexibleExtensionList](./core.inferflexibleextensionlist.md) | Get the extension types from a list of PrioritizedExtensions. |
|  [MapCommandToAction](./core.mapcommandtoaction.md) | A utility type which maps the passed in extension command in an action that is called via <code>manager.data.actions.commandName()</code>. |
|  [MapHelpers](./core.maphelpers.md) | A utility type which maps the passed in extension helpers to a method called with <code>manager.data.helpers.helperName()</code>. |
|  [MarkNames](./core.marknames.md) | A utility type for retrieving the name of an extension only when it's a mark extension. |
|  [NameOfExtension](./core.nameofextension.md) | Utility type for retrieving the name of an extension. |
|  [NodeNames](./core.nodenames.md) | A utility type for retrieving the name of an extension only when it's a node extension. |
|  [OptionsOfExtension](./core.optionsofextension.md) | Utility type for retrieving the extension options from an extension. |
|  [PlainNames](./core.plainnames.md) | A utility type for retrieving the name of an extension only when it's a plain extension. |
|  [ProsemirrorTypeOfExtension](./core.prosemirrortypeofextension.md) | Utility type for retrieving the prosemirror type of the extension which can be:<!-- -->- <code>NodeType</code> - <code>MarkType</code> - <code>never</code> |
|  [SchemaFromExtensions](./core.schemafromextensions.md) | Gets the schema from a list of extensions |
|  [TypeOfExtensionClass](./core.typeofextensionclass.md) | Retrieve the instance type from an ExtensionClass. |
|  [TypeOfExtensionClassList](./core.typeofextensionclasslist.md) | This is a utility type which allows for retrieving the instance types from an array of Extensions. |

