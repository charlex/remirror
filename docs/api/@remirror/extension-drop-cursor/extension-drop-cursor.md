<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@remirror/extension-drop-cursor](./extension-drop-cursor.md)

## extension-drop-cursor package

## Classes

|  Class | Description |
|  --- | --- |
|  [DropCursorExtension](./extension-drop-cursor.dropcursorextension.md) | This is a custom drop cursor that allows for customising what is shown in the document with a rendered view. |
|  [DropCursorState](./extension-drop-cursor.dropcursorstate.md) |  |

## Functions

|  Function | Description |
|  --- | --- |
|  [dropCursorPlugin(params, extension)](./extension-drop-cursor.dropcursorplugin.md) | Create a drop cursor plugin which adds a decoration to the position that is currently being dragged over. |

## Interfaces

|  Interface | Description |
|  --- | --- |
|  [DropCursorExtensionComponentProps](./extension-drop-cursor.dropcursorextensioncomponentprops.md) |  |
|  [DropCursorExtensionOptions](./extension-drop-cursor.dropcursorextensionoptions.md) |  |

## Type Aliases

|  Type Alias | Description |
|  --- | --- |
|  [DropCursorType](./extension-drop-cursor.dropcursortype.md) | This indicates whether the current cursor position is within a textblock or between two nodes. |

