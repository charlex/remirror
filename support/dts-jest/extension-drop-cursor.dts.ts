import { HelpersFromExtensions } from '@remirror/core';
import { DropCursorExtension } from '@remirror/extension-drop-cursor';

type DropCursorExtensionHelpers = HelpersFromExtensions<DropCursorExtension>;
const cursorHelpers: DropCursorExtensionHelpers = Object.create(null);

// @dts-jest:pass:snap
cursorHelpers.isDragging();
