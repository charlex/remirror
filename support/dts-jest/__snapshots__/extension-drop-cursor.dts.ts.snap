// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cursorHelpers.isDragging() (type) should match snapshot 1`] = `"boolean"`;
